### PR TITLE
Fix error when prefab generator skips generation

### PR DESCRIPTION
**Pull request by amzn-mike**
## What does this PR do?

When a prefab group exists in the manifest but generation is skipped due to a registry setting, it leaves behind an empty prefab group which causes exporting to fail later on since the group still exists.  This fix causes the group to be removed before skipping the entry and adds an info message to notify users that a registry setting caused generation to be skipped.

Fixes https://github.com/o3de/o3de/issues/15361

## How was this PR tested?

Followed repro steps in linked bug and verified asset processes correctly after fix.

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

**Comment by amzn-mike** [comment]
I added some unit tests to verify this fix: https://github.com/o3de/o3de/pull/15385

I'll merge them in after this fix is merged

**Pull request merged**
